rclcpp/test/rclcpp/test_message_memory_strategy.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <memory>
+
+#include "rclcpp/message_memory_strategy.hpp"
+#include "test_msgs/msg/empty.hpp"
+
+TEST(TestMemoryStrategies, construct_destruct) {
+  rclcpp::message_memory_strategy::MessageMemoryStrategy<test_msgs::msg::Empty> memory_strategy1;
+
+  auto allocator = std::make_shared<std::allocator<void>>();
+  rclcpp::message_memory_strategy::MessageMemoryStrategy<test_msgs::msg::Empty> memory_strategy2(
+    allocator);
+}
+
+TEST(TestMemoryStrategies, standard_allocation) {
+  rclcpp::message_memory_strategy::MessageMemoryStrategy<test_msgs::msg::Empty> memory_strategy;
+  auto default_message = memory_strategy.create_default();
+  ASSERT_NE(nullptr, default_message);
```

Comment:
I may be misunderstanding this, but it looks to me like `create_default` is actually a factory method that returns a `MessageMemoryStrategy`.  So I'd think a better test here would be:

```
auto memory_strategy = rclcpp::message_memory_strategy::MessageMemoryStrategy<test_msgs::msg::Empty>::create_default();
```

Reply:
You are correct about that. Fixed the test.